File: backend/services/post-service/internal/model/post.go
// Package model defines the data models for the post service.
package model

import (
	"time"
)

// Post represents a community post.
type Post struct {
	ID               string    `gorm:"type:uuid;primaryKey;default:gen_random_uuid()" json:"id"`
	AuthorID         string    `gorm:"type:uuid;not null;index" json:"author_id"`
	AuthorNickname   string    `gorm:"type:varchar(100);not null" json:"author_nickname"`
	AuthorAvatar     string    `gorm:"type:text;default:''" json:"author_avatar"`
	Content          string    `gorm:"type:text;not null" json:"content"`
	MoodTag          string    `gorm:"type:varchar(50);not null;index" json:"mood_tag"`
	Images           string    `gorm:"type:jsonb;default:'[]'" json:"images"` // JSON array of URLs
	VoiceURL         string    `gorm:"type:text" json:"voice_url"`
	StickerHug       int32     `gorm:"default:0" json:"sticker_hug"`
	StickerCheer     int32     `gorm:"default:0" json:"sticker_cheer"`
	StickerUnderstand int32    `gorm:"default:0" json:"sticker_understand"`
	CommentCount     int32     `gorm:"default:0;index" json:"comment_count"`
	HasAIReply       bool      `gorm:"default:false" json:"has_ai_reply"`
	IsReported       bool      `gorm:"default:false" json:"is_reported"`
	IsHidden         bool      `gorm:"default:false" json:"is_hidden"`
	CreatedAt        time.Time `gorm:"autoCreateTime;index" json:"created_at"`
	UpdatedAt        time.Time `gorm:"autoUpdateTime" json:"updated_at"`
}

// TableName overrides the default table name.
func (Post) TableName() string {
	return "posts"
}

File: frontend/lib/domain/models/post.dart
/// Post model matching the backend Post entity.
class Post {
  final String id;
  final String authorId;
  final String authorNickname;
  final String authorAvatar;
  final String content;
  final String moodTag;
  final List<String> images;
  final String? voiceUrl;
  final int stickerHug;
  final int stickerCheer;
  final int stickerUnderstand;
  final int commentCount;
  final bool hasAiReply;
  final DateTime createdAt;

  const Post({
    required this.id,
    required this.authorId,
    required this.authorNickname,
    this.authorAvatar = '',
    required this.content,
    required this.moodTag,
    this.images = const [],
    this.voiceUrl,
    this.stickerHug = 0,
    this.stickerCheer = 0,
    this.stickerUnderstand = 0,
    this.commentCount = 0,
    this.hasAiReply = false,
    required this.createdAt,
  });

  factory Post.fromJson(Map<String, dynamic> json) {
    return Post(
      id: json['id'] as String,
      authorId: json['author_id'] as String,
      authorNickname: json['author_nickname'] as String,
      authorAvatar: json['author_avatar'] as String? ?? '',
      content: json['content'] as String,
      moodTag: json['mood_tag'] as String,
      images: (json['images'] as List<dynamic>?)
              ?.map((e) => e.toString())
              .toList() ??
          [],
      voiceUrl: json['voice_url'] as String?,
      stickerHug: json['sticker_hug'] as int? ?? 0,
      stickerCheer: json['sticker_cheer'] as int? ?? 0,
      stickerUnderstand: json['sticker_understand'] as int? ?? 0,
      commentCount: json['comment_count'] as int? ?? 0,
      hasAiReply: json['has_ai_reply'] as bool? ?? false,
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }
}
